### PR TITLE
getFutureEstimate Function

### PR DESCRIPTION
**Pull request by kdimitrov03**
Added a method that estimates the animals and the space they will need in the shelter in a certain amount of years

### CODE REVIEW

File: src/hierarchy/Animal.ts [hunk 1]
```diff
@@ -87,6 +87,12 @@ export default abstract class Animal implements ICanWalk {
     }
     return true
   }
+  public grow(): void {
+    this.age += 1
+    if (this.age >= AGE_TO_BE_ADULT) {
```
Comment (rkoychev) [inline src/hierarchy/Animal.ts:92]: maybe  age needs a setter that does this check for adult, so we dont need to pass adult or not but setting the check takes care of that on its own

File: tests/animalShelter.test.ts [hunk 2]
```diff
@@ -106,29 +106,45 @@ describe('Animal Shelter Tests', () => {
...
+  const tortoise2 = new Tortoise('t2', 4, false)
+  const squirrel = new Squirrel('s1', 3, false, 'Maple', 44, 4)
+  const squirrel2 = new Squirrel('s2', 4, false, 'Maple', 44, 4)
+  const lion = new Lion('simba', 4, true)
+  animalShelter.addAnimal(tortoise)
+  animalShelter.addAnimal(tortoise2)
+  animalShelter.addAnimal(squirrel)
+  animalShelter.addAnimal(squirrel2)
+  animalShelter.addAnimal(lion)
+  const estimate = animalShelter.getFutureEstimate(2)
```
Comment (rkoychev) [inline tests/animalShelter.test.ts:149]: what are  we checking here ? i dont see anything being tested.    Make additional tests and try to cover all corner cases

**Review by rkoychev — CHANGES_REQUESTED** [review]

File: tests/animalShelter.test.ts (hunk 2 above)
Comment (kdimitrov03) [reply tests/animalShelter.test.ts:149]: Yeah not really sure how to do so because we are using random and the estimate that is returned changes every time

Comment (rkoychev) [reply tests/animalShelter.test.ts]: If you have such issue , dont make a pr but ask colleagues for help or even better do some research on the topic. I see you found a way to do this already, but in the future don't let out pr's that are not finished without asking explicitly for help on the troublesome topic

File: tests/animalShelter.test.ts [hunk 3]
```diff
@@ -167,4 +178,56 @@ describe('Animal Shelter Tests', () => {
...
+    animalShelter.addAnimal(tortoise2)
+    animalShelter.addAnimal(squirrel)
+    animalShelter.addAnimal(lion)
+    const animals = animalShelter.report()
+    expect(animals.anaimalsWhoClimbTrees).toBe(4)
+    expect(animals.animalsWhoJump).toBe(2)
+    expect(animals.animalsWhoRun).toBe(1)
+  })
+
+  test('should return an estimate of future space needed if females always give birth', () => {
```
Comment (rkoychev) [inline tests/animalShelter.test.ts:197]: we have an explicit requirement to report for 5 years in the future, yet we don't even have a single test on this functionality

**Review by rkoychev — CHANGES_REQUESTED** [review]

**Review by rkoychev — APPROVED** [review]
